### PR TITLE
CORDA-3193: Update docs to mention branching strategy

### PR DESCRIPTION
**Pull request by lockathan**
Removed mention of master branch except to describe that the master branch is no longer used.
Added explanation of branching strategy to contributing guide.
Renamed building-against-master to building-against-non-release

### CODE REVIEW

**Comment by gendal** [comment]
for an external contributor who may not know the answers to these questions, what do we suggest?  open against the default branch? contact us first?

File: docs/source/contributing.rst [hunk 1]
```diff
@@ -47,11 +47,26 @@ Making the required changes
...
+
+Selecting a development branch
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Corda does not use the `master` branch for development. Instead, all development work is done on the branch named 
+according to the release that it represents. The following pattern is used for branch naming:
+`release/os/{major version}.{minor version}`
+
+Note that `release` is always part of the name of the branch, even for unreleased versions of Corda. The default 
+github branch will always be the current development branch of Corda. Development work should target the oldest version
```
Comment (blsemo) [inline docs/source/contributing.rst]: As @gendal said - mention here that this isn't just the default GitHub branch, it should also be the default to work against (unless fixing a specific issue in a specific version).

File: docs/source/contributing.rst [hunk 2]
```diff
@@ -117,7 +132,7 @@ Merging the changes back into Corda
 -----------------------------------
 You should merge the changes back into Corda as follows:
 
-1. Create a pull request from your fork to the ``master`` branch of the Corda repo
+1. Create a pull request from your fork to the current development branch of the Corda repo
```
Comment (blsemo) [inline docs/source/contributing.rst]: What is the `current development branch`? Shouldn't that be the release branch for the version you developed against?

**Review by blsemo — CHANGES_REQUESTED** [review]

File: docs/source/contributing.rst [hunk 3]
```diff
@@ -105,7 +121,7 @@ You should document any changes to Corda's public API as follows:
    with a full stop. We also start comments with capital letters, even for inline comments. Where Java APIs have
    synonyms (e.g. ``%d`` and ``%date``), we prefer the longer form for legibility reasons. You can configure your IDE
    to highlight these in bright yellow
-2. Update the relevant `.rst file(s) <https://github.com/corda/corda/tree/master/docs/source>`_
+2. Update the relevant `.rst file(s)
```
Comment (edward-prosser) [inline docs/source/contributing.rst]: Remove the backtick from before .rst

**Review by blsemo — APPROVED** [review]
LGTM

**Pull request merged**